File: Majora's Math/majorasmath.java
import java.util.Scanner;

public class majorasmath {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        int h = scanner.nextInt();
        int n = scanner.nextInt();

        int a = scanner.nextInt();
        int b = scanner.nextInt();
        int c = scanner.nextInt();
        int d = scanner.nextInt();

        for (int i = 0; i < n; i++) {
            String arrowType = scanner.next();

            if (arrowType.equals("standard")) {
                h -= a;
            } else if (arrowType.equals("fire")) {
                h -= b;
            } else if (arrowType.equals("ice")) {
                h -= c;
            } else {
                h -= d;
            }
        }

        if (h <= 0) {
            System.out.println("dead");
        } else {
            System.out.println(h);
        }
    }
}

File: Majora's Math/majorasmath.c
#include <stdio.h>
#include <string.h>

int main() {
    int h, n;
    scanf("%d %d", &h, &n);

    int a, b, c, d;
    scanf("%d %d %d %d", &a, &b, &c, &d);

    for (int i = 0; i < n; i++) {
        char arrowType[9];
        scanf("%s", arrowType);

        if (strcmp(arrowType, "standard") == 0) {
            h -= a;
        } else if (strcmp(arrowType, "fire") == 0) {
            h -= b;
        } else if (strcmp(arrowType, "ice") == 0) {
            h -= c;
        } else {
            h -= d;
        }
    }

    if (h <= 0) {
        printf("dead\n");
    } else {
        printf("%d\n", h);
    }

    return 0;
}
